bin/test-db

```diff
@@ -8,9 +8,23 @@ from argparse import RawTextHelpFormatter
 
 # singerio images have required keyfile
 image_name = "singerio/mongo"
-image_tag = "4.2-bionic"
-# image_tag = "4.4-bionic"
-# image_tag = "5.0"
+
+# image_tag = "4.2-bionic"  # version 5.0.15 TODO Remove?
+# image_tag = "4.4-bionic"  # version 4.4.6
+# image_tag = "5.0"         # version 5.0.15
+image_tag = "6.0"           # version 6.0.4
+
+# organize command options based on image_tag
+command_options = {
+    '4.2-bionic': {'conf_option': 'mongo',
+                    'conn_option': 'mongo:4.2.0-bionic mongo'},
+    '4.4-bionic': {'conf_option': 'mongo',
+                   'conn_option': 'mongo:4.4.0-bionic mongo'},
```

Comment:
This causes us to have to download two docker images to run this script:
- `singerio/mongo:4.4-bionic` to run `test_db start`
- `mongo:4.4.0-bionic` to run `test_db connect`

## Running `start`
```
$ ./bin/test-db start
Starting Docker process mongo1 using command:
    sudo docker run -e "MONGO_INITDB_ROOT_USERNAME=stitch_dev" -e "MONGO_INITDB_ROOT_PASSWORD=REDACTED"         -p 27017:27017 --name mongo1         -d singerio/mongo:4.4-bionic         --auth         --keyFile /opt/mongo/keyfile --replSet rs0

Unable to find image 'singerio/mongo:4.4-bionic' locally
4.4-bionic: Pulling from singerio/mongo
e7ae86ffe2df: Pull complete
cb44957d0c54: Pull complete
1b034681f705: Pull complete
f68a0696c1b2: Pull complete
e7e03afd9141: Pull complete
a4a217eed0e5: Pull complete
af2e1e83b32e: Pull complete
1a263184825b: Pull complete
8cadecd5d9b5: Pull complete
a0b5b7c565dc: Pull complete
637a52a3fee1: Pull complete
79955afdd883: Pull complete
a091888e6afc: Pull complete
9750b5cf679b: Pull complete
Digest: sha256:4779a0dbc12aac2f09d406b61eabbee9d3896c6c085c28e0efc126d4edda1e4f
Status: Downloaded newer image for singerio/mongo:4.4-bionic
1b40934fc86ff8a48fe254188e87477e01187b59ebcfd82ef2bbd0f3c7769aa1
Process started successfully. Starting Oplog replication.
Retrieving IP addr of mongodb container
172.17.0.3
Initiate replSet using:
     docker exec mongo1 mongo --host 172.17.0.3 test -u stitch_dev -p REDACTED --authenticationDatabase admin --eval 'rs.initiate({_id: "rs0", members: [{_id: 0, host: "127.0.0.1:27017"}]})'

MongoDB shell version v4.4.6
connecting to: mongodb://172.17.0.3:27017/test?authSource=admin&compressors=disabled&gssapiServiceName=mongodb
Implicit session: session { "id" : UUID("63fd6b72-8141-4432-b99e-e38f483c910b") }
MongoDB server version: 4.4.6
{ "ok" : 1 }
Oplog configured correctly.
```

## Running `connect`
```
$ ./bin/test-db connect
Retrieving IP addr of mongodb container
172.17.0.3
Attempting to connect to running container using a mongo container
docker run -it --rm mongo:4.4.0-bionic mongo --host 172.17.0.3 test -u stitch_dev -p REDACTED --authenticationDatabase admin
Unable to find image 'mongo:4.4.0-bionic' locally
4.4.0-bionic: Pulling from library/mongo
f08d8e2a3ba1: Pull complete
3baa9cb2483b: Pull complete
94e5ff4c0b15: Pull complete
1860925334f9: Pull complete
9d42806c06e6: Pull complete
31a9fd218257: Pull complete
5bd6e3f73ab9: Pull complete
f6ae7a64936b: Pull complete
80fde2cb25c5: Pull complete
1bec62fe62fc: Pull complete
2cf4970a1653: Pull complete
39fac3226e16: Pull complete
86bca9c64faf: Pull complete
Digest: sha256:b7de11e3dc4a8fd787e4e1cb7a7fc49e5015fe649ba82090a2d74280425b9c3e
Status: Downloaded newer image for mongo:4.4.0-bionic
MongoDB shell version v4.4.0
connecting to: mongodb://172.17.0.3:27017/test?authSource=admin&compressors=disabled&gssapiServiceName=mongodb
Implicit session: session { "id" : UUID("657810e0-0356-4f2d-b239-e015e8b5d2e0") }
MongoDB server version: 4.4.6
Welcome to the MongoDB shell.
For interactive help, type "help".
For more comprehensive documentation, see
        https://docs.mongodb.com/
Questions? Try the MongoDB Developer Community Forums
        https://community.mongodb.com
---
The server generated these startup warnings when booting:
        2023-10-09T18:06:56.761+00:00: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine. See http://dochub.mongodb.org/core/prodnotes-filesystem
---
---
        Enable MongoDB's free cloud-based monitoring service, which will then receive and display
        metrics about your deployment (disk utilization, CPU, operation statistics, etc).

        The monitoring data will be available on a MongoDB website with a unique URL accessible to you
        and anyone you share the URL with. MongoDB may use this information to make product
        improvements and to suggest MongoDB products and deployment options to you.

        To enable free monitoring, run the following command: db.enableFreeMonitoring()
        To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
---
rs0:PRIMARY>
```

Reply:
Updated